src/Database/Concerns/ResourceSyncing.php

```diff
@@ -43,4 +46,15 @@ public function shouldSync(): bool
     {
         return true;
     }
+
+    public function tenants(): MorphToMany
+    {
+        return $this->morphToMany($this->getResourceTenantModelName(), 'tenant_resources', 'tenant_resources', 'resource_global_id', 'tenant_id', 'global_id')
+            ->using(TenantMorphPivot::class);
+    }
+
+    public function getResourceTenantModelName(): string // todo better name
+    {
+        return config('tenancy.tenant_model', Tenant::class);
+    }
```

Comment:
Why was this added again?

Reply:
In case the developer wants to provide a Tenant model other than the default one. A classic example is in our tests https://github.com/archtechx/tenancy/pull/997/files#diff-ff1377ee24fda8c295959b05c5156fda8ae36d131e56f20e220f153c4d19e8ccR181-R184

Reply:
The link doesn't work now - can you provide a permalink to some commit/lines of code, instead of the PR diff?

Reply:
https://github.com/archtechx/tenancy/blob/ea3e90987b40c4025246d6e80b7487971d9c36e5/tests/ResourceSyncingUsingPolymorphicTest.php#L290-L293

Reply:
A few things:
1. What's the actual point of this? Can't we just change the config in the tests?
2. Are tests the only use case or is there any other use case?
3. As we've discussed several times before, the term "resource tenant" doesn't really make sense.

Reply:
> What's the actual point of this? Can't we just change the config in the tests?

We can. But if the developer wants to provide a different tenant model in real app. 

> Are tests the only use case or is there any other use case?

I believe, there can be a real app case. I'm not sure what, but I believe there should be an option to provide a different tenant model. 

> As we've discussed several times before, the term "resource tenant" doesn't really make sense.

Yes, but there is already a method `getTenantModelName`, that's why the current name was feeling right.

Reply:
> What if the developer wants to provide a different tenant model in a real app

I don't see why the developer would want to use a different model **just for this one thing**, but use the configured Tenant model for everything else. Am I missing something?

> I believe there should be an option to provide a different tenant model

In all other parts of the package we use the exact same tenant model — the configured one.

Reply:
> In all other parts of the package we use the exact same tenant model — the configured one.

I see, that makes sense.